### PR TITLE
Add back VsCoq in CI

### PR DESCRIPTION
**Pull request by maximedenes**
VsCoq's main branch now relies on Coq master

### CODE REVIEW

**Comment by SkySkimmer** [comment]
@coqbot run full ci

**Comment by maximedenes** [comment]
I may have to add some ocaml dependencies for vscoq (not sure). If so, where should they go? There seems to be several variables defining dependencies in the `Dockerfile`.

**Comment by SkySkimmer** [comment]
You're using the base switch so BASE_ONLY_OPAM

**Comment by gares** [comment]
I think serapi has mode deps then VSCoq, you ca use that opam switch.

If you want to add yarn, then I guess we need a new docker image indeed.

File: dev/ci/ci-vscoq.sh [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+
+ci_dir="$(dirname "$0")"
+. "${ci_dir}/ci-common.sh"
+
+git_download vscoq
+
+if [ "$DOWNLOAD_ONLY" ]; then exit 0; fi
+
+( cd "$CI_BUILD_DIR/vscoq/language-server"
+  dune build && dune runtest
```
Comment (SkySkimmer) [inline dev/ci/ci-vscoq.sh:13]: Please use separate lines

Also I think you need to use -p to get dune to work when called in a subdirectory of coq

**Comment by maximedenes** [comment]
@coqbot run full CI

**Comment by maximedenes** [comment]
> I think serapi has more deps then VSCoq, you can use that opam switch.

Unfortunately, the edge switch seems to use `ppx_deriving_yojson` whereas VsCoq uses `ppx_yojson_conv`, as recommended:

"ppx_yojson_conv is a more recent deriving extension for Yojson that uses a more durable technical foundation and is more actively maintained. We keep maintaing ppx_deriving_yojson for our existing users, but we would recommend that new projects start from ppx_yojson_conv instead"

I'll add it as a dependency.

**Comment by maximedenes** [comment]
@coqbot run full CI

**Comment by maximedenes** [comment]
@coqbot run full CI

**Comment by maximedenes** [comment]
@coqbot run full CI

**Comment by maximedenes** [comment]
@coqbot run full CI

**Comment by maximedenes** [comment]
@coqbot run full CI

**Comment by maximedenes** [comment]
@SkySkimmer would you mind taking this PR?

**Review by SkySkimmer — APPROVED** [review]

**Comment by SkySkimmer** [comment]
@coqbot merge now

**Comment by maximedenes** [comment]
Thanks!

**Pull request merged**